### PR TITLE
🗣️ feat: Azure OpenAI speech

### LINKED ISSUE #3094: Enhancement: Azure OpenAI STT/TTS
### What features would you like to see added?

I would love to configure TTS with an Azure Endpoint. I tried the following without success:
```yaml
                - group: "azure-openai" # arbitrary name
                  plugins: true
                  assistants: true
                  apiKey: "${AZURE_CFG_API_KEY}"
                  instanceName: "robads-ai-schweden"  # name of the resource group or instance
                  version: "2024-02-15-preview"
                  models:
                    gpt-3.5-turbo-1106:
                      deploymentName: gpt-35-turbo
                    gpt-4-vision-preview:
                      deploymentName: gpt-4-vision-preview
                    gpt-4o:
                      deploymentName: gpt-4o
                    tts:
                      deploymentName: tts-hd
```

### More details

I would love to use Azure OpenAI as TTS Endpoint

### Which components are impacted by your request?

Endpoints

### Pictures

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by berry-13**
## Summary

fixed Azure OpenAI STT and added Azure OpenAI TTS

## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.
- [x] A pull request for updating the documentation has been submitted.

### CODE REVIEW

**Comment by danny-avila** [comment]
slated for v0.7.4

File: api/server/services/Files/Audio/textToSpeech.js [hunk 1]
```diff
@@ -228,6 +282,9 @@ async function ttsRequest(provider, ttsSchema, { input, voice, stream = true } =
     case 'openai':
       [url, data, headers] = openAIProvider(ttsSchema, input, voice);
       break;
+    case 'azureOpenAI':
```
Comment (danny-avila) [inline api/server/services/Files/Audio/textToSpeech.js]: can you use enums for this?

File: api/server/services/Files/Audio/getVoices.js [hunk 2]
```diff
@@ -28,6 +28,9 @@ async function getVoices(req, res) {
       case 'openai':
         voices = ttsSchema.openai?.voices;
         break;
+      case 'azureOpenAI':
```
Comment (danny-avila) [inline api/server/services/Files/Audio/getVoices.js]: enums

**Review by danny-avila — CHANGES_REQUESTED** [review]
Also, can you raise updated docs for librechat.ai with this change?

**Comment by berry-13** [comment]
> Also, can you raise updated docs for librechat.ai with this change?

the docs are in the same [PR](https://github.com/LibreChat-AI/librechat.ai/pull/61) of #2921

**Comment by danny-avila** [comment]
@berry-13 please resolve the merge conflicts (result of #2921)

**Review by danny-avila — APPROVED** [review]

**Pull request merged**